bindings/python/options.py

```diff
@@ -1,44 +1,51 @@
-"""A module for parsing standard options intended to be used by Python
-storage plugins."""
+"""A module for converting option to storage plugins between
+valid percent-encoded URL query strings and and dicts.
 
-import json
+A URL query string is a string of the form
 
+    key1=value1;key2=value2...
 
-class Options(dict):
-    """A dict representation of the options string `options`.
+where the keys and and values are percent-encoded.  The key-value
+pairs may be separated by either semicolon (;) or ampersand (&).
+
+Percent-encoding means that all characters that are digits, letters or
+one of "~._-" are encoded as-is, while all other is encoded as their
+unicode byte number in hex with each byte preceeded "%". For example
+"a" would be encoded as "a", "+" would be encoded as "%2B" and "å" as
+"%C3%A5".
+
+If a value starts with "%%", the rest of the value is assumed to be a
+percent-encoded json strings.
+"""
+
+import json
+import re
 
-    Options is a string of the form
+import dlite
 
-        opt1=value1;opt2=value2...
 
-    where semicolon (;) may be replaced with an ampersand (&).
+class Options(dict):
+    """A dict representation of the options string `options` with
+    attribute access.
 
-    Default values may be provided via the `defaults` argument.  It
-    should either be a dict or a string of the same form as `options`.
+    Arguments:
```

Comment:
Arguments? Instance Variables I suppose? And shouldn't this be in the docstring og the __init__?

Reply:
Yes, you are right, the arguments refer to the arguments passed to the `__init__()` method. But it is common to document arguments in the class docstring. I like to do it for the following reasons:
- In case the class has both a `__new__()` and a `__init__()` method, we don't need to duplicate the argument documentation if we keep it in the class docstring.
- The `__new__()` and `__init__()` methods starts with an underscore and may be hidden by some documentation systems.